lib/kube/proxy/forwarder.go

```diff
@@ -498,6 +509,12 @@ const accessDeniedMsg = "[00] access denied"
 
 // authenticate function authenticates request
 func (f *Forwarder) authenticate(req *http.Request) (*authContext, error) {
+	// If the cluster is not licensed for Kubernetes, return an error to the client.
+	if !f.cfg.ClusterFeatures().Kubernetes {
+		// If the cluster is not licensed for Kubernetes, return an error to the client
+		// but in a way that the code is translated to 500 error code.
```

Comment:
If this isn't too much hassle, it might be worth testing that the error is truly translated to a 500 error code.

Reply:
Added in 9a249562a6963d76981bdeb3e9939f04378c308f

Reply:
Should this be a 500 or a 403 (Forbidden)?

Reply:
We seem to return `trace.AccessDenied` (which becomes a 403 in http) whenever some license feature check fails, which seems appropriate here. Is there a kube-specific reason why it should not be returned as an access denied?

Reply:
I preferred 500 to distinguish from cases where the user does not have access but reverted to Forbidden